image/helpers.go

```diff
@@ -213,19 +213,26 @@ type DownloadOptions struct {
 	TargetDir string
 	Channel   string
 	CohortKey string
+	Basename  string
 }
 
-var errRevisionAndCohort = errors.New("cannot specify both revision and cohort")
+var (
+	errRevisionAndCohort = errors.New("cannot specify both revision and cohort")
+	errPathInBase        = errors.New("cannot specify a path in basename (use target dir for that)")
+)
 
 func (opts *DownloadOptions) validate() error {
-	if opts.Revision.Unset() || opts.CohortKey == "" {
-		return nil
+	if strings.ContainsRune(opts.Basename, filepath.Separator) {
```

Comment:
Should this also complain if Basename ends with `.snap`? If so then cmd_download.go needs an update too.